extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.tracking.js

```diff
@@ -1,35 +1,151 @@
 define('wikia.articleVideo.featuredVideo.tracking', [], function () {
+	var state = getDefaultState();
+	var gaCategory = 'featured-video';
+	var playerInstance;
+	var wasAlreadyUnmuted = false;
+	var depth = 0;
+
 	function getDefaultState() {
 		return {
-			wasAlreadyUnmuted: false
+			wasStartTracked: false,
+			progress: {
+				durationTracked: 0,
+				percentTracked: 0
+			}
 		}
 	}
 
 	function track(gaData) {
-		console.log(gaData);
+		if (!gaData.label) {
+			throw new Error('No tracking label provided');
+		}
+
+		var finalGAData = {
+			action: gaData.action || 'click',
+			category: gaCategory,
+			label: gaData.label,
+			//value tracks sound state: 1 for muted, 0 for unmuted
+			value: Number(playerInstance.getMute())
+		};
+
+		// Will be replaced by connecting Internal + GA trackers
+		console.info(finalGAData);
 	}
 
-	var state = getDefaultState();
+	return function (providedPlayerInstance, willAutoplay, providedGACategory) {
+		playerInstance = providedPlayerInstance;
+		gaCategory = providedGACategory;
 
-	return function (playerInstance) {
-		playerInstance.on('play', function () {
+		playerInstance.once('ready', function () {
 			track({
-				label: 'featured-video-play'
+				label: 'load',
+				action: 'impression'
+			});
+
+			var relatedPlugin = playerInstance.getPlugin('related');
+
+			relatedPlugin.on('open', function () {
+				track({
+					label: 'recommended-video',
+					action: 'impression'
+				});
+
+				state = getDefaultState();
+			});
+
+			relatedPlugin.on('play', function (data) {
+				depth++;
+
+				var labelPrefix = data.auto ? 'recommended-video-autoplay-' : 'recommended-video-select-';
+
+				track({
+					label: labelPrefix + data.position,
+					action: 'impression'
```

Comment:
consider using a var for that, since it's used 4-5 times

Reply:
We will use GA predefined categories once we connect GA tracker: https://wikia-inc.atlassian.net/browse/XW-4089